lightning/src/ln/channelmanager.rs

```diff
@@ -4046,6 +4048,34 @@ where
 			)
 	}
 
+	#[cfg(async_payments)]
+	fn initiate_async_payment(
+		&self, invoice: &StaticInvoice, payment_id: PaymentId
+	) -> Result<(), InvoiceError> {
+		if invoice.message_paths().is_empty() { return Err(Bolt12SemanticError::MissingPaths.into()) }
+
+		let reply_path = self.create_blinded_path(Some(payment_id))
+			.map_err(|_| Bolt12SemanticError::MissingPaths)?;
+
+		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(self);
+		let payment_release_secret = self.pending_outbound_payments.static_invoice_received(
+			invoice, payment_id, &*self.entropy_source
+		).map_err(|e| InvoiceError::from_string(format!("{:?}", e)))?;
```

Comment:
Could we keep the translation to `InvoiceError` in the handler? It should determine when one should be sent, IMO. Maybe (some of) this should go right in the handler? We also shouldn't respond if we don't know the payment id or the offer doesn't match to avoid correlating payments.